src/main/java/it/gov/pagopa/payment/service/payment/common/CommonPreAuthServiceImpl.java

```diff
@@ -91,43 +94,37 @@ public AuthPaymentDTO previewPayment(TransactionInProgress trx, String channel)
   protected void checkPreAuth(String trxCode, String userId, TransactionInProgress trx) {
     String walletStatus = walletConnector.getWallet(trx.getInitiativeId(), userId).getStatus();
     if (PaymentConstants.WALLET_STATUS_SUSPENDED.equals(walletStatus)){
-      throw new ClientExceptionWithBody(
-              HttpStatus.FORBIDDEN,
+      throw new UserSuspendedException(
               PaymentConstants.ExceptionCode.USER_SUSPENDED_ERROR,
               "The user has been suspended for initiative [%s]".formatted(trx.getInitiativeId()));
     }
 
     if (PaymentConstants.WALLET_STATUS_UNSUBSCRIBED.equals(walletStatus)){
-      throw new ClientExceptionWithBody(
-              HttpStatus.FORBIDDEN,
+      throw new UserNotOnboardedException(
               PaymentConstants.ExceptionCode.USER_UNSUBSCRIBED,
               "The user has unsubscribed from initiative [%s]".formatted(trx.getInitiativeId()));
     }
 
     if (trx.getTrxDate().plusMinutes(authorizationExpirationMinutes).isBefore(OffsetDateTime.now())) {
-      throw new ClientExceptionWithBody(
-              HttpStatus.NOT_FOUND,
+      throw new TransactionNotFoundOrExpiredException(
               PaymentConstants.ExceptionCode.TRX_NOT_FOUND_OR_EXPIRED,
               "Cannot find transaction with trxCode [%s]".formatted(trxCode));
     }
 
     if (trx.getUserId() != null && !userId.equals(trx.getUserId())) {
-      throw new ClientExceptionWithBody(
-              HttpStatus.FORBIDDEN,
+      throw new UserNotAllowedException(
               PaymentConstants.ExceptionCode.TRX_ANOTHER_USER,
               "Transaction with trxCode [%s] is already assigned to another user".formatted(trxCode));
     }
 
     if(SyncTrxStatus.AUTHORIZED.equals(trx.getStatus())){
-      throw new ClientExceptionWithBody(
-              HttpStatus.FORBIDDEN,
+      throw new TransactionAlreadyAuthorizedException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor